upstairs/src/lib.rs

```diff
@@ -1740,6 +1755,51 @@ impl Upstairs {
 
         Ok(())
     }
+
+    /*
+     * Complete a downstairs operation.
+     *
+     * Returns true if the guest should be notified.
+     */
+    fn complete(
```

Comment:
It's my fault.
We have too many things named complete.  I'm going to try and do a better job of naming things without so much overlap.  Nothing for this review, I'm really just leaving this comment as a shame reminder to myself to go fix the problem I have created.